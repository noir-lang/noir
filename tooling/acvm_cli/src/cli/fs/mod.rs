pub(super) mod inputs;
pub(super) mod witness;
