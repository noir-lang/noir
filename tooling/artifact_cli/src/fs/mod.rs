pub mod artifact;
pub mod inputs;
pub mod witness;
