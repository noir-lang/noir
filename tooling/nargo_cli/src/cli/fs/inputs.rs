use noirc_abi::{
    input_parser::{Format, InputValue},
    Abi, InputMap, MAIN_RETURN_NAME,
};
use std::{collections::BTreeMap, path::Path};

use crate::errors::FilesystemError;

/// Returns the circuit's parameters and its return value, if one exists.
/// # Examples
///
/// ```ignore
/// let (input_map, return_value): (InputMap, Option<InputValue>) =
///   read_inputs_from_file(path, "Verifier", Format::Toml, &abi)?;
/// ```
pub(crate) fn read_inputs_from_file<P: AsRef<Path>>(
    path: P,
    file_name: &str,
    format: Format,
    abi: &Abi,
) -> Result<(InputMap, Option<InputValue>), FilesystemError> {
    if abi.is_empty() {
        return Ok((BTreeMap::new(), None));
    }

    let file_path = path.as_ref().join(file_name).with_extension(format.ext());
    if !file_path.exists() {
        return Err(FilesystemError::MissingTomlFile(file_name.to_owned(), file_path));
    }

    let input_string = std::fs::read_to_string(file_path).unwrap();
    let mut input_map = format.parse(&input_string, abi)?;
    let return_value = input_map.remove(MAIN_RETURN_NAME);

    Ok((input_map, return_value))
}
