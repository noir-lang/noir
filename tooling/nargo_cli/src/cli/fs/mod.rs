use std::{
    fs::File,
    io::Write,
    path::{Path, PathBuf},
};

pub(super) mod inputs;
pub(super) mod program;
pub(super) mod witness;

pub(super) fn create_named_dir(named_dir: &Path, name: &str) -> PathBuf {
    std::fs::create_dir_all(named_dir)
        .unwrap_or_else(|_| panic!("could not create the `{name}` directory"));

    PathBuf::from(named_dir)
}

pub(super) fn write_to_file(bytes: &[u8], path: &Path) -> String {
    let display = path.display();

    let mut file = match File::create(path) {
        Err(why) => panic!("couldn't create {display}: {why}"),
        Ok(file) => file,
    };

    match file.write_all(bytes) {
        Err(why) => panic!("couldn't write to {display}: {why}"),
        Ok(_) => display.to_string(),
    }
}
