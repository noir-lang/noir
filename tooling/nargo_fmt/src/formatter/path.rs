use noirc_frontend::{
    ast::{Path, PathKind, UnresolvedType},
    token::{Keyword, Token},
};

use super::Formatter;

impl<'a> Formatter<'a> {
    pub(super) fn format_path(&mut self, path: Path) {
        self.skip_comments_and_whitespace();

        match path.kind {
            PathKind::Plain => (),
            PathKind::Crate => {
                self.write_keyword(Keyword::Crate);
                self.write_token(Token::DoubleColon);
            }
            PathKind::Dep => {
                self.write_keyword(Keyword::Dep);
                self.write_token(Token::DoubleColon);
            }
            PathKind::Super => {
                self.write_keyword(Keyword::Super);
                self.write_token(Token::DoubleColon);
            }
        }

        for (index, segment) in path.segments.into_iter().enumerate() {
            if index > 0 {
                self.write_token(Token::DoubleColon);
            }
            self.write_identifier(segment.ident);

            if let Some(generics) = segment.generics {
                self.format_turbofish(generics);
            }
        }
    }

    pub(super) fn format_turbofish(&mut self, generics: Vec<UnresolvedType>) {
        self.write_token(Token::DoubleColon);
        self.write_token(Token::Less);
        for (index, typ) in generics.into_iter().enumerate() {
            if index > 0 {
                self.write_comma();
                self.write_space();
            }
            self.format_type(typ);
        }

        // Skip trailing comma, if any
        self.skip_comments_and_whitespace();
        if self.is_at(Token::Comma) {
            self.bump();
        }

        self.write_token(Token::Greater);
    }
}
