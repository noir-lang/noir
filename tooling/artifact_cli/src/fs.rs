use std::path::Path;

use color_eyre::eyre::{self, Context};
use noirc_artifacts::{contract::ContractArtifact, program::ProgramArtifact};

/// A parsed JSON build artifact.
#[derive(Debug, Clone)]
pub enum Artifact {
    Program(ProgramArtifact),
    Contract(ContractArtifact),
}

impl Artifact {
    /// Try to parse an artifact as a binary program or a contract
    pub fn read_from_file(path: &Path) -> eyre::Result<Artifact> {
        let json = std::fs::read(path)
            .with_context(|| format!("failed to read artifact from '{}'", path.display()))?;

        let as_program = || serde_json::from_slice::<ProgramArtifact>(&json).map(Artifact::Program);
        let as_contract =
            || serde_json::from_slice::<ContractArtifact>(&json).map(Artifact::Contract);

        Ok(as_program().or_else(|e| as_contract().map_err(|_| e))?)
    }
}
