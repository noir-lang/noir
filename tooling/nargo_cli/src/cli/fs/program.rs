use std::path::{Path, PathBuf};

use nargo::artifacts::{contract::ContractArtifact, program::ProgramArtifact};
use noirc_frontend::graph::CrateName;

use crate::errors::FilesystemError;

use super::{create_named_dir, write_to_file};

pub(crate) fn save_program_to_file<P: AsRef<Path>>(
    program_artifact: &ProgramArtifact,
    crate_name: &CrateName,
    circuit_dir: P,
) -> PathBuf {
    let circuit_name: String = crate_name.into();
    save_build_artifact_to_file(program_artifact, &circuit_name, circuit_dir)
}

pub(crate) fn save_contract_to_file<P: AsRef<Path>>(
    compiled_contract: &ContractArtifact,
    circuit_name: &str,
    circuit_dir: P,
) -> PathBuf {
    save_build_artifact_to_file(compiled_contract, circuit_name, circuit_dir)
}

fn save_build_artifact_to_file<P: AsRef<Path>, T: ?Sized + serde::Serialize>(
    build_artifact: &T,
    artifact_name: &str,
    circuit_dir: P,
) -> PathBuf {
    create_named_dir(circuit_dir.as_ref(), "target");
    let circuit_path = circuit_dir.as_ref().join(artifact_name).with_extension("json");

    write_to_file(&serde_json::to_vec(build_artifact).unwrap(), &circuit_path);

    circuit_path
}

pub(crate) fn read_program_from_file<P: AsRef<Path>>(
    circuit_path: P,
) -> Result<ProgramArtifact, FilesystemError> {
    let file_path = circuit_path.as_ref().with_extension("json");

    let input_string =
        std::fs::read(&file_path).map_err(|_| FilesystemError::PathNotValid(file_path))?;
    let program = serde_json::from_slice(&input_string)
        .map_err(|err| FilesystemError::ProgramSerializationError(err.to_string()))?;

    Ok(program)
}
