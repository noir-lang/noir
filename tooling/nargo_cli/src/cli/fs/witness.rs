use std::path::{Path, PathBuf};

use acvm::acir::native_types::WitnessStack;
use nargo::constants::WITNESS_EXT;

use super::{create_named_dir, write_to_file};
use crate::errors::FilesystemError;

pub(crate) fn save_witness_to_dir<P: AsRef<Path>>(
    witness_stack: WitnessStack,
    witness_name: &str,
    witness_dir: P,
) -> Result<PathBuf, FilesystemError> {
    create_named_dir(witness_dir.as_ref(), "witness");
    let witness_path = witness_dir.as_ref().join(witness_name).with_extension(WITNESS_EXT);

    let buf: Vec<u8> = witness_stack.try_into()?;

    write_to_file(buf.as_slice(), &witness_path);

    Ok(witness_path)
}
