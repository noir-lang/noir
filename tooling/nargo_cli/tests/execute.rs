#[allow(unused_imports)]
#[cfg(test)]
mod tests {
    // Some of these imports are consumed by the injected tests
    use assert_cmd::prelude::*;
    use predicates::prelude::*;

    use std::collections::BTreeMap;
    use std::fs;
    use std::path::PathBuf;
    use std::process::Command;

    use super::*;

    test_binary::build_test_binary_once!(mock_backend, "../backend_interface/test-binaries");

    // include tests generated by `build.rs`
    include!(concat!(env!("OUT_DIR"), "/execute.rs"));
}
