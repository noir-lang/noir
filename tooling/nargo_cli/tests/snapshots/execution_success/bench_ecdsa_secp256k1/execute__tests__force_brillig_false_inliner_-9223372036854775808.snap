---
source: tooling/nargo_cli/tests/execute.rs
expression: artifact
---
{
  "noir_version": "[noir_version]",
  "hash": "[hash]",
  "abi": {
    "parameters": [
      {
        "name": "hashed_message",
        "type": {
          "kind": "array",
          "length": 32,
          "type": {
            "kind": "integer",
            "sign": "unsigned",
            "width": 8
          }
        },
        "visibility": "private"
      },
      {
        "name": "pub_key_x",
        "type": {
          "kind": "array",
          "length": 32,
          "type": {
            "kind": "integer",
            "sign": "unsigned",
            "width": 8
          }
        },
        "visibility": "private"
      },
      {
        "name": "pub_key_y",
        "type": {
          "kind": "array",
          "length": 32,
          "type": {
            "kind": "integer",
            "sign": "unsigned",
            "width": 8
          }
        },
        "visibility": "private"
      },
      {
        "name": "signature",
        "type": {
          "kind": "array",
          "length": 64,
          "type": {
            "kind": "integer",
            "sign": "unsigned",
            "width": 8
          }
        },
        "visibility": "private"
      }
    ],
    "return_type": null,
    "error_types": {}
  },
  "bytecode": [
    "func 0",
    "current witness index : 160",
    "private parameters indices : [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159]",
    "public parameters indices : []",
    "return value indices : []",
    "BLACKBOX::RANGE [(_0, 8)] []",
    "BLACKBOX::RANGE [(_1, 8)] []",
    "BLACKBOX::RANGE [(_2, 8)] []",
    "BLACKBOX::RANGE [(_3, 8)] []",
    "BLACKBOX::RANGE [(_4, 8)] []",
    "BLACKBOX::RANGE [(_5, 8)] []",
    "BLACKBOX::RANGE [(_6, 8)] []",
    "BLACKBOX::RANGE [(_7, 8)] []",
    "BLACKBOX::RANGE [(_8, 8)] []",
    "BLACKBOX::RANGE [(_9, 8)] []",
    "BLACKBOX::RANGE [(_10, 8)] []",
    "BLACKBOX::RANGE [(_11, 8)] []",
    "BLACKBOX::RANGE [(_12, 8)] []",
    "BLACKBOX::RANGE [(_13, 8)] []",
    "BLACKBOX::RANGE [(_14, 8)] []",
    "BLACKBOX::RANGE [(_15, 8)] []",
    "BLACKBOX::RANGE [(_16, 8)] []",
    "BLACKBOX::RANGE [(_17, 8)] []",
    "BLACKBOX::RANGE [(_18, 8)] []",
    "BLACKBOX::RANGE [(_19, 8)] []",
    "BLACKBOX::RANGE [(_20, 8)] []",
    "BLACKBOX::RANGE [(_21, 8)] []",
    "BLACKBOX::RANGE [(_22, 8)] []",
    "BLACKBOX::RANGE [(_23, 8)] []",
    "BLACKBOX::RANGE [(_24, 8)] []",
    "BLACKBOX::RANGE [(_25, 8)] []",
    "BLACKBOX::RANGE [(_26, 8)] []",
    "BLACKBOX::RANGE [(_27, 8)] []",
    "BLACKBOX::RANGE [(_28, 8)] []",
    "BLACKBOX::RANGE [(_29, 8)] []",
    "BLACKBOX::RANGE [(_30, 8)] []",
    "BLACKBOX::RANGE [(_31, 8)] []",
    "BLACKBOX::RANGE [(_32, 8)] []",
    "BLACKBOX::RANGE [(_33, 8)] []",
    "BLACKBOX::RANGE [(_34, 8)] []",
    "BLACKBOX::RANGE [(_35, 8)] []",
    "BLACKBOX::RANGE [(_36, 8)] []",
    "BLACKBOX::RANGE [(_37, 8)] []",
    "BLACKBOX::RANGE [(_38, 8)] []",
    "BLACKBOX::RANGE [(_39, 8)] []",
    "BLACKBOX::RANGE [(_40, 8)] []",
    "BLACKBOX::RANGE [(_41, 8)] []",
    "BLACKBOX::RANGE [(_42, 8)] []",
    "BLACKBOX::RANGE [(_43, 8)] []",
    "BLACKBOX::RANGE [(_44, 8)] []",
    "BLACKBOX::RANGE [(_45, 8)] []",
    "BLACKBOX::RANGE [(_46, 8)] []",
    "BLACKBOX::RANGE [(_47, 8)] []",
    "BLACKBOX::RANGE [(_48, 8)] []",
    "BLACKBOX::RANGE [(_49, 8)] []",
    "BLACKBOX::RANGE [(_50, 8)] []",
    "BLACKBOX::RANGE [(_51, 8)] []",
    "BLACKBOX::RANGE [(_52, 8)] []",
    "BLACKBOX::RANGE [(_53, 8)] []",
    "BLACKBOX::RANGE [(_54, 8)] []",
    "BLACKBOX::RANGE [(_55, 8)] []",
    "BLACKBOX::RANGE [(_56, 8)] []",
    "BLACKBOX::RANGE [(_57, 8)] []",
    "BLACKBOX::RANGE [(_58, 8)] []",
    "BLACKBOX::RANGE [(_59, 8)] []",
    "BLACKBOX::RANGE [(_60, 8)] []",
    "BLACKBOX::RANGE [(_61, 8)] []",
    "BLACKBOX::RANGE [(_62, 8)] []",
    "BLACKBOX::RANGE [(_63, 8)] []",
    "BLACKBOX::RANGE [(_64, 8)] []",
    "BLACKBOX::RANGE [(_65, 8)] []",
    "BLACKBOX::RANGE [(_66, 8)] []",
    "BLACKBOX::RANGE [(_67, 8)] []",
    "BLACKBOX::RANGE [(_68, 8)] []",
    "BLACKBOX::RANGE [(_69, 8)] []",
    "BLACKBOX::RANGE [(_70, 8)] []",
    "BLACKBOX::RANGE [(_71, 8)] []",
    "BLACKBOX::RANGE [(_72, 8)] []",
    "BLACKBOX::RANGE [(_73, 8)] []",
    "BLACKBOX::RANGE [(_74, 8)] []",
    "BLACKBOX::RANGE [(_75, 8)] []",
    "BLACKBOX::RANGE [(_76, 8)] []",
    "BLACKBOX::RANGE [(_77, 8)] []",
    "BLACKBOX::RANGE [(_78, 8)] []",
    "BLACKBOX::RANGE [(_79, 8)] []",
    "BLACKBOX::RANGE [(_80, 8)] []",
    "BLACKBOX::RANGE [(_81, 8)] []",
    "BLACKBOX::RANGE [(_82, 8)] []",
    "BLACKBOX::RANGE [(_83, 8)] []",
    "BLACKBOX::RANGE [(_84, 8)] []",
    "BLACKBOX::RANGE [(_85, 8)] []",
    "BLACKBOX::RANGE [(_86, 8)] []",
    "BLACKBOX::RANGE [(_87, 8)] []",
    "BLACKBOX::RANGE [(_88, 8)] []",
    "BLACKBOX::RANGE [(_89, 8)] []",
    "BLACKBOX::RANGE [(_90, 8)] []",
    "BLACKBOX::RANGE [(_91, 8)] []",
    "BLACKBOX::RANGE [(_92, 8)] []",
    "BLACKBOX::RANGE [(_93, 8)] []",
    "BLACKBOX::RANGE [(_94, 8)] []",
    "BLACKBOX::RANGE [(_95, 8)] []",
    "BLACKBOX::RANGE [(_96, 8)] []",
    "BLACKBOX::RANGE [(_97, 8)] []",
    "BLACKBOX::RANGE [(_98, 8)] []",
    "BLACKBOX::RANGE [(_99, 8)] []",
    "BLACKBOX::RANGE [(_100, 8)] []",
    "BLACKBOX::RANGE [(_101, 8)] []",
    "BLACKBOX::RANGE [(_102, 8)] []",
    "BLACKBOX::RANGE [(_103, 8)] []",
    "BLACKBOX::RANGE [(_104, 8)] []",
    "BLACKBOX::RANGE [(_105, 8)] []",
    "BLACKBOX::RANGE [(_106, 8)] []",
    "BLACKBOX::RANGE [(_107, 8)] []",
    "BLACKBOX::RANGE [(_108, 8)] []",
    "BLACKBOX::RANGE [(_109, 8)] []",
    "BLACKBOX::RANGE [(_110, 8)] []",
    "BLACKBOX::RANGE [(_111, 8)] []",
    "BLACKBOX::RANGE [(_112, 8)] []",
    "BLACKBOX::RANGE [(_113, 8)] []",
    "BLACKBOX::RANGE [(_114, 8)] []",
    "BLACKBOX::RANGE [(_115, 8)] []",
    "BLACKBOX::RANGE [(_116, 8)] []",
    "BLACKBOX::RANGE [(_117, 8)] []",
    "BLACKBOX::RANGE [(_118, 8)] []",
    "BLACKBOX::RANGE [(_119, 8)] []",
    "BLACKBOX::RANGE [(_120, 8)] []",
    "BLACKBOX::RANGE [(_121, 8)] []",
    "BLACKBOX::RANGE [(_122, 8)] []",
    "BLACKBOX::RANGE [(_123, 8)] []",
    "BLACKBOX::RANGE [(_124, 8)] []",
    "BLACKBOX::RANGE [(_125, 8)] []",
    "BLACKBOX::RANGE [(_126, 8)] []",
    "BLACKBOX::RANGE [(_127, 8)] []",
    "BLACKBOX::RANGE [(_128, 8)] []",
    "BLACKBOX::RANGE [(_129, 8)] []",
    "BLACKBOX::RANGE [(_130, 8)] []",
    "BLACKBOX::RANGE [(_131, 8)] []",
    "BLACKBOX::RANGE [(_132, 8)] []",
    "BLACKBOX::RANGE [(_133, 8)] []",
    "BLACKBOX::RANGE [(_134, 8)] []",
    "BLACKBOX::RANGE [(_135, 8)] []",
    "BLACKBOX::RANGE [(_136, 8)] []",
    "BLACKBOX::RANGE [(_137, 8)] []",
    "BLACKBOX::RANGE [(_138, 8)] []",
    "BLACKBOX::RANGE [(_139, 8)] []",
    "BLACKBOX::RANGE [(_140, 8)] []",
    "BLACKBOX::RANGE [(_141, 8)] []",
    "BLACKBOX::RANGE [(_142, 8)] []",
    "BLACKBOX::RANGE [(_143, 8)] []",
    "BLACKBOX::RANGE [(_144, 8)] []",
    "BLACKBOX::RANGE [(_145, 8)] []",
    "BLACKBOX::RANGE [(_146, 8)] []",
    "BLACKBOX::RANGE [(_147, 8)] []",
    "BLACKBOX::RANGE [(_148, 8)] []",
    "BLACKBOX::RANGE [(_149, 8)] []",
    "BLACKBOX::RANGE [(_150, 8)] []",
    "BLACKBOX::RANGE [(_151, 8)] []",
    "BLACKBOX::RANGE [(_152, 8)] []",
    "BLACKBOX::RANGE [(_153, 8)] []",
    "BLACKBOX::RANGE [(_154, 8)] []",
    "BLACKBOX::RANGE [(_155, 8)] []",
    "BLACKBOX::RANGE [(_156, 8)] []",
    "BLACKBOX::RANGE [(_157, 8)] []",
    "BLACKBOX::RANGE [(_158, 8)] []",
    "BLACKBOX::RANGE [(_159, 8)] []",
    "BLACKBOX::ECDSA_SECP256K1 [(_32, 8), (_33, 8), (_34, 8), (_35, 8), (_36, 8), (_37, 8), (_38, 8), (_39, 8), (_40, 8), (_41, 8), (_42, 8), (_43, 8), (_44, 8), (_45, 8), (_46, 8), (_47, 8), (_48, 8), (_49, 8), (_50, 8), (_51, 8), (_52, 8), (_53, 8), (_54, 8), (_55, 8), (_56, 8), (_57, 8), (_58, 8), (_59, 8), (_60, 8), (_61, 8), (_62, 8), (_63, 8), (_64, 8), (_65, 8), (_66, 8), (_67, 8), (_68, 8), (_69, 8), (_70, 8), (_71, 8), (_72, 8), (_73, 8), (_74, 8), (_75, 8), (_76, 8), (_77, 8), (_78, 8), (_79, 8), (_80, 8), (_81, 8), (_82, 8), (_83, 8), (_84, 8), (_85, 8), (_86, 8), (_87, 8), (_88, 8), (_89, 8), (_90, 8), (_91, 8), (_92, 8), (_93, 8), (_94, 8), (_95, 8), (_96, 8), (_97, 8), (_98, 8), (_99, 8), (_100, 8), (_101, 8), (_102, 8), (_103, 8), (_104, 8), (_105, 8), (_106, 8), (_107, 8), (_108, 8), (_109, 8), (_110, 8), (_111, 8), (_112, 8), (_113, 8), (_114, 8), (_115, 8), (_116, 8), (_117, 8), (_118, 8), (_119, 8), (_120, 8), (_121, 8), (_122, 8), (_123, 8), (_124, 8), (_125, 8), (_126, 8), (_127, 8), (_128, 8), (_129, 8), (_130, 8), (_131, 8), (_132, 8), (_133, 8), (_134, 8), (_135, 8), (_136, 8), (_137, 8), (_138, 8), (_139, 8), (_140, 8), (_141, 8), (_142, 8), (_143, 8), (_144, 8), (_145, 8), (_146, 8), (_147, 8), (_148, 8), (_149, 8), (_150, 8), (_151, 8), (_152, 8), (_153, 8), (_154, 8), (_155, 8), (_156, 8), (_157, 8), (_158, 8), (_159, 8), (_0, 8), (_1, 8), (_2, 8), (_3, 8), (_4, 8), (_5, 8), (_6, 8), (_7, 8), (_8, 8), (_9, 8), (_10, 8), (_11, 8), (_12, 8), (_13, 8), (_14, 8), (_15, 8), (_16, 8), (_17, 8), (_18, 8), (_19, 8), (_20, 8), (_21, 8), (_22, 8), (_23, 8), (_24, 8), (_25, 8), (_26, 8), (_27, 8), (_28, 8), (_29, 8), (_30, 8), (_31, 8)] [_160]",
    "EXPR [ (1, _160) -1 ]",
    "",
    ""
  ],
  "debug_symbols": "bY1bCoMwFET3cr/zYbRKyVZKkTzlQkhCHoUS3HujVGrBzzPDmamgtCjLjM74BOxRwXrJM3rXqAKduj1MgbuNU+YxA6O3OwHtFLB+GFcCBq0GNnbrkzSFXin9T5mGP6WBiGgtLvP5u8UvHpELq79oipOnNr/D0Rx+iF5qVaLelvauzX8A",
  "file_map": {
    "50": {
      "source": "use dep::std;\n\nfn main(hashed_message: [u8; 32], pub_key_x: [u8; 32], pub_key_y: [u8; 32], signature: [u8; 64]) {\n    let valid_signature =\n        std::ecdsa_secp256k1::verify_signature(pub_key_x, pub_key_y, signature, hashed_message);\n    assert(valid_signature);\n}\n",
      "path": ""
    }
  },
  "names": [
    "main"
  ],
  "brillig_names": []
}
